File: python/programmers/greedy/p4.py
from collections import deque


def solution(people, limit):
    people.sort()
    people = deque(people)
    ans = 0

    while people:
        ans += 1
        weight = 0
        cnt = 0
        while cnt < 2 and len(people) != 0:
            if people[0] + weight > limit:
                break
            else:
                weight += people.popleft()
            cnt += 1

    return ans

File: java/programmers/src/greedy/p4.java
package greedy;

import java.util.ArrayDeque;
import java.util.Arrays;
import java.util.Deque;

public class p4 {

	public static void main(String[] args) {
		
		int[] people = {70, 80, 50,40};
		
		int limit = 100;
		int ans=0;
		
		Arrays.sort(people);
	
		
		int i=0;
		int heavy = people.length-1;
		
		while (i<heavy) {
			if( people[i] + people[heavy] <=limit) {
				i++;
				ans++;
			}
			heavy--;
		}
		
		System.out.println();
	
	}

}
